reference/test/stop/residual_norm_kernels.cpp

```diff
@@ -50,54 +50,324 @@ namespace {
 
 
 template <typename T>
-class ResidualNormReduction : public ::testing::Test {
+class ResidualNorm : public ::testing::Test {
 protected:
     using Mtx = gko::matrix::Dense<T>;
     using NormVector = gko::matrix::Dense<gko::remove_complex<T>>;
 
-    ResidualNormReduction()
+    ResidualNorm()
     {
         exec_ = gko::ReferenceExecutor::create();
-        factory_ = gko::stop::ResidualNormReduction<T>::build()
+        factory_ = gko::stop::ResidualNorm<T>::build()
                        .with_reduction_factor(r<T>::value)
                        .on(exec_);
```

Comment:
Could you rename this factory, so it is clear that it is using `mode::rhs_norm`? I was a bit confused in the beginning.